File: initial/src/main/java/hello/SimpleBookRepository.java
package hello;

public class SimpleBookRepository implements BookRepository {

    @Override
    public Book getByIsbn(String isbn) {
        simulateSlowService();
        return new Book(isbn, "Some book");
    }

    // Don't do this at home
    private void simulateSlowService() {
        try {
            long time = (long) (5000L);
            Thread.sleep(time);
        } catch (InterruptedException e) {
            throw new IllegalStateException(e);
        }
    }

}

File: initial-kotlin/src/main/kotlin/com/example/caching/SimpleBookRepository.kt
package com.example.caching

import org.springframework.stereotype.Component

@Component
class SimpleBookRepository : BookRepository {

    override fun getByIsbn(isbn: String): Book {
        simulateSlowService()
        return Book(isbn, "Some book")
    }

    // Don't do this at home
    private fun simulateSlowService() {
        Thread.sleep(3000L)
    }
}
